hpx/runtime/serialization/variant.hpp

```diff
@@ -0,0 +1,78 @@
+//  Copyright (c) 2015 Anton Bikineev
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_SERIALIZATION_VARIANT_HPP
+#define HPX_SERIALIZATION_VARIANT_HPP
+
+#include <hpx/runtime/serialization/serialize.hpp>
+
+#include <boost/variant.hpp>
+
+namespace hpx { namespace serialization
+{
+    namespace detail
+    {
+        struct variant_save_visitor: boost::static_visitor<>
+        {
+            variant_save_visitor(output_archive& ar)
+              : ar(ar)
+            {}
+
+            template <class T>
+            void operator()(const T& value)
+            {
+                ar << value;
+            }
+
+        private:
+            output_archive& ar;
+        };
+
+        template <class... Args>
+        struct load_helper_t{};
+
+        template <class Variant, class... Tail>
+        BOOST_FORCEINLINE void load_helper(input_archive& ar, Variant& var,
+                int which, load_helper_t<boost::detail::variant::void_, Tail...>)
```

Comment:
K-ballo: I couldn't find any macro for this. We can also use mpl list defined inside variant specialization instead of variadic templates, how it's done in boost/serialization/variant.hpp. For this case it would be easier just to copy that file with a few modifications. What do you think about it?